<launch>
<arg name="video_device" default="/dev/video0" />
<arg name="image_width" default="1280" />
<arg name="image_height" default="720" />

<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="log">
<param name="video_device" value="$(arg video_device)" />
<param name="image_width" value="$(arg image_width)" />
<param name="image_height" value="$(arg image_height)"/>
<param name="pixel_format" value="mjpeg" />
<param name="camera_frame_id" value="usb_cam" />
<param name="io_method" value="mmap"/>
</node>

<arg name="marker1Id"         default="0"/>
<arg name="marker2Id"         default="1"/>
<arg name="marker1Size"        default="0.157"/>    
<arg name="marker2Size"        default="0.125"/>    
    <arg name="eye"               default="left"/>
    <arg name="dct_normalization" default="True" />
    <arg name="dct_filter_size"   default="2" />
    <arg name="marker1_frame"     default="marker_ref_frame" />
    <arg name="marker2_frame"     default="marker_rob_frame" />


<node pkg="aruco_ros" type="single" name="aruco_simple0">    
<remap from="/camera_info" to="/usb_cam/camera_info" />
<remap from="/image" to="/usb_cam/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size" value="$(arg marker1Size)"/>
        <param name="marker_id" value="$(arg marker1Id)"/>
        <param name="marker_frame" value="$(arg marker1_frame)" />
<param name="camera_frame" value="cam" />
</node>

<node pkg="aruco_ros" type="single" name="aruco_simple1">    
<remap from="/camera_info" to="/usb_cam/camera_info" />
<remap from="/image" to="/usb_cam/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size" value="$(arg marker2Size)"/>
        <param name="marker_id" value="$(arg marker2Id)"/>
        <param name="marker_frame" value="$(arg marker2_frame)" />
<param name="camera_frame" value="cam" />
</node>

<node name="aruco_tf2_broadcaster" pkg="arduino_robot" type="aruco_tf2_broadcaster.py" respawn="false" output="screen" >
</node>

</launch>
